src/main/java/software/amazon/encryption/s3/materials/MaterialsDescription.java

```diff
@@ -20,92 +19,119 @@ public class MaterialsDescription implements Map<String, String> {
-  public void putAll(Map<? extends String, ? extends String> m) {
+    @Override
+    public String put(String key, String value) {
       throw new UnsupportedOperationException("This map is immutable");
-  }
-
-  @Override
-  public void clear() {
-      materialsDescription.clear();
-  }
-
-  @Override
-  public Set<String> keySet() {
-     return materialsDescription.keySet();
-  }
-
-  @Override
-  public Collection<String> values() {
-      return materialsDescription.values();
-  }
-
-  @Override
-  public Set<Entry<String, String>> entrySet() {
-      return materialsDescription.entrySet();
-  }
-
-
-  public static class Builder {
-        private final Map<String, String> materialsDescription = new HashMap<>();
-        public Builder put(String key, String value) {
-            if (key == null || value == null) {
-                throw new IllegalArgumentException("Key and value must not be null");
+    }
+
+    @Override
+    public String remove(Object key) {
+      return materialsDescription.remove(key);
+    }
+
+    @Override
+    public void putAll(Map<? extends String, ? extends String> m) {
+        throw new UnsupportedOperationException("This map is immutable");
+    }
+
+    @Override
+    public void clear() {
+        materialsDescription.clear();
+    }
+
+    @Override
+    public Set<String> keySet() {
+       return materialsDescription.keySet();
+    }
+
+    @Override
+    public Collection<String> values() {
+        return materialsDescription.values();
+    }
+
+    @Override
+    public Set<Entry<String, String>> entrySet() {
+        return materialsDescription.entrySet();
+    }
+
+    /**
+     * Builder for MaterialsDescription.
+     */
+    public static class Builder {
+          private final Map<String, String> materialsDescription = new HashMap<>();
+
+          /**
+           * @param key the key to add
+           * @param value the value to add
+           * @return a reference to this object so that method calls can be chained together.
+           * @throws IllegalArgumentException if key or value is null
+           */
+          public Builder put(String key, String value) {
+              if (key == null || value == null) {
+                  throw new IllegalArgumentException("Key and value must not be null");
+              }
+              materialsDescription.put(key, value);
+              return this;
+          }
+
+          /**
+           * @param description the map of key-value pairs to add
+           * @return a reference to this object so that method calls can be chained together.
+           * @throws IllegalArgumentException if description is null
+           */
+          public Builder putAll(Map<String, String> description) {
+            if (description == null) {
+              throw new IllegalArgumentException("Description must not be null");
             }
-            materialsDescription.put(key, value);
+            materialsDescription.putAll(description);
```

Comment:
Hmmm these changes should have been merged in from the `ReEncrypt-feature` branch, I think.